project/src/main/java/com/google/sps/servlets/CompetitionInfoServlet.java

```diff
@@ -95,7 +96,14 @@ private UserCompetition getUserCompetition(Connection conn, int userId, int comp
     List<CompetitorInfo> participants = getCompetitionParticipants(conn, competitionId);
     CompetitorInfo user = getCompetitorInfo(conn, userId, competitionId);
 
-    return new UserCompetition(competitionId, competitionName, creatorId, creatorEmail, start, end, user, participants);
+    int rank = 0;
+    for (int i = 1; i <= participants.size(); i++) {
```

Comment:
might be better to put this for loop in a seperate function

Reply:
{nit} so in the if statement instead of "rank = i" you can just "return i" and don't have to go through the whole loop

Reply:
ye sorted